vnext/Microsoft.ReactNative/Views/DevMenu.cpp

```diff
@@ -114,26 +113,31 @@ void DevMenuManager::CreateAndShowUI() noexcept {
       Mso::React::ReactOptions::UseFastRefresh(m_context->Properties()) ? L"Disable Fast Refresh"
                                                                         : L"Enable Fast Refresh");
 
-#if defined(USE_HERMES)
-  devMenu.SamplingProfilerText().Text(
-      !Microsoft::ReactNative::HermesSamplingProfiler::IsStarted() ? L"Start Hermes sampling profiler"
-                                                                   : L"Stop Hermes sampling profiler");
-  std::ostringstream os;
-  if (Microsoft::ReactNative::HermesSamplingProfiler::IsStarted()) {
-    os << "Sampling profiler running..";
-  } else {
-    os << "Click to start.";
-  }
+  if (Mso::React::ReactOptions::GetJsiEngine(m_context->Properties()) == Mso::React::JSIEngine::Hermes) {
+      devMenu.SamplingProfilerText().Text(
+          !Microsoft::ReactNative::HermesSamplingProfiler::IsStarted() ? L"Start Hermes sampling profiler"
+                                                                       : L"Stop Hermes sampling profiler");
+      devMenu.SamplingProfilerIcon().Glyph(
+          !Microsoft::ReactNative::HermesSamplingProfiler::IsStarted() ? winrt::to_hstring("\uE1E5")
+                                                                       : winrt::to_hstring("\uE15B"));
```

Comment:
you can just use L"\ue1e5" and L"\ue15b" and the compiler will construct the hstring for you from them (the L prefix creates a utf16 character string)